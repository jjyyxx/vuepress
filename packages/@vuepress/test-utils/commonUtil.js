import { fs, path } from '@vuepress/shared-utils'
import LRU from 'lru-cache'

const cache = new LRU({ max: 1000 })

async function getFragment (dirname, name) {
  let content = cache.get(name)
  if (content) {
    return content
  }
  const target = path.resolve(dirname, `fragments/${name}.md`)
  content = await fs.readFile(target, 'utf-8')
  cache.set(name, content)
  return content
}

module.exports = {
  getFragment
}
