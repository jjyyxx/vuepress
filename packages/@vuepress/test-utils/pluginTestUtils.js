const { datatypes: { isFunction }} = require('@vuepress/shared-utils')
const { getContextSingleton } = require('./contextUtil')

class Plugin {
  constructor (...args) {
    this.args = args
  }

  async init () {
    const [plugin, options, context] = this.args
    this.context = context || (await getContextSingleton())
    this.plugin = isFunction(plugin)
      ? plugin(options, this.context)
      : plugin
    return this
  }

  extendMarkdown () {
    this.plugin.extendMarkdown(this.markdown)
    return this
  }

  get markdown () {
    return this.context.markdown
  }
}

const use = (...options) => new Plugin(...options)

module.exports = {
  use,
  Plugin
}
