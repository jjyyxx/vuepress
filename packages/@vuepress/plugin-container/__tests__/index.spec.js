import { use } from '@vuepress/test-utils/pluginTestUtils.js'
import { getFragment } from '@vuepress/test-utils/commonUtil.js'
import containerPlugin from '..'

describe('containers', async () => {
  const containerLabels = ['tip', 'tip-override', 'warning', 'danger', 'v-pre']
  const plugin = await use(containerPlugin, { type: 'tip' }).init()

  plugin.extendMarkdown()
  containerLabels.forEach(label => {
    test(label, async () => {
      const input = await getFragment(__dirname, `container-${label}`)
      const output = plugin.markdown.render(input)
      expect(output).toMatchSnapshot()
    })
  })
})
