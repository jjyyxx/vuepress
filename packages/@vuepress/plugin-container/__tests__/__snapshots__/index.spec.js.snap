// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers danger 1`] = `
Object {
  "data": Object {
    "__data_block": Object {},
  },
  "dataBlockString": "",
  "html": <p>::: danger I am a danger :::
</p>,
}
`;

exports[`containers tip 1`] = `
Object {
  "data": Object {
    "__data_block": Object {},
  },
  "dataBlockString": "",
  "html": <p>::: tip I am a tip :::
</p>,
}
`;

exports[`containers tip-override 1`] = `
Object {
  "data": Object {
    "__data_block": Object {},
  },
  "dataBlockString": "",
  "html": <p>::: tip 提示 I am a tip :::
</p>,
}
`;

exports[`containers v-pre 1`] = `
Object {
  "data": Object {
    "__data_block": Object {},
  },
  "dataBlockString": "",
  "html": <p>::: v-pre I am a v-pre :::
</p>,
}
`;

exports[`containers warning 1`] = `
Object {
  "data": Object {
    "__data_block": Object {},
  },
  "dataBlockString": "",
  "html": <p>::: warning I am a warning :::
</p>,
}
`;
